### DISCUSSION TITLE
Arguments from Event [Question]

### DISCUSSION THREAD

**Discussion by Davidm16** [flet-dev/flet#1686] [Q&A] [answered]
### Question

Hi, I have not found information on how to pass arguments from an Event like Tkinter. If I set all the widgets as variable and a def for each the code is three times larger, can you apply this feature?

### Code sample

```python
# Flet
ft.IconButton(ft.icons.REFRESH, on_click=refrescar),
# Tkinter
Button(window, text="<-", command= lambda: cambiaVentana(0))
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by rkleivel** [comment, marked as answer]
Something like this?:
```python
import flet as ft

def main(page: ft.Page):
    def refrescar(my_arg=None):
        print(my_arg)
        # perform your refresh actions

    page.add(ft.IconButton(ft.icons.REFRESH, on_click=lambda _: refrescar(my_arg=123)))

ft.app(target=main)
```

**Reply by Davidm16** [reply]
Yes! Thank you
